pkg/flashdb/include/fal_cfg.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2022 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_flashdb
+ * @{
+ *
+ * @file
+ * @brief       Flash Abstraction Layer partition configuration
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+
+#ifndef FAL_CFG_H
+#define FAL_CFG_H
+
+#include "board.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Partition table is defined at compile time (not read from flash)
+ */
+#define FAL_PART_HAS_TABLE_CFG
+
+/**
+ * @brief FAL <-> MTD adapter
+ */
+extern struct fal_flash_dev mtd_flash0;
+
+/**
+ * @brief   flash device table
+ */
+#define FAL_FLASH_DEV_TABLE                                          \
+{                                                                    \
+    &mtd_flash0,                                                     \
+}
+
+/**
+ * @brief Partition 0
+ */
+#ifdef FAL_PART0_LABEL
```

Comment:
Given that there is only one MTD device, is this a dummy partition table, and if so, why 4 partitions?

Reply:
But we have only one partition by the above, and a brief search in upstream sources shows that there don't have to be 4 entries.

Reply:
A user can specify up to 4 partitions before having to touch this file :wink:

Reply:
I thought that by https://github.com/RIOT-OS/RIOT/pull/17612/files#r897214173 we can only have one backend ... maybe it'd help adding a few words on what partitions mean here.

Reply:
One backend can have multiple partitions.

Reply:
So multiple partitions are usable, but because FlashDB expects them all to be provided on a single continuous address (rather than in a (partition, address) addressing), any and all must be on the same MTD device. Got it now, but it'd be good to have this information accessible to someone using the module after it's merged, along with a pointer to where the choice of MTD partition goes when initializing the database.

Reply:
Like this? 7db5c1e084be3c4070bd3a7dde7cb96803817fde